### DISCUSSION TITLE
[PROPOSAL] Drop the netCDF 2.3.2 API and tools

### DISCUSSION THREAD

**Discussion by derobins** [HDFGroup/hdf4#410] [Proposed Change]
### **PROBLEM**

HDF4 provides a netCDF interface compatible with netCDF 2.3.2 (circa the early 90s), along with implementation of the ncdump and ncgen tools.

These API calls and tools will clash with their modern equivalents in applications that use both HDF4 and netCDF. We've added configure options to hack around this problem, but these elderly API calls and tools should be removed if they are no longer being used by the community.

### **PROPOSED FIX**

* Remove the netCDF 2.3.2 public API (essentially the functionality in `mfhdf/libsrc/netcdf.h.in`)
* Remove the netCDF tools: ncdump and ncgen
* Remove the netCDF nctest test program
* Internal netCDF API calls would probably remain unless there was a reason to strip them out

### **HOW WOULD USERS BE AFFECTED?**

**PRO**
➕ Eliminate HDF4 / netCDF clashes without configure hacks
➕ Slim down the library, removing a lot of low-quality code

**CON**
➖ People who need the netCDF 2.3.2 API or tools would be sad

**NEUTRAL**
✔️ The netCDF functionality could be maintained in a 4.2 maintenance branch (though I am not excited about maintaining multiple branches of HDF4)

**Comment by hyoklee** [comment]
Test netcdf's [HDF4 support](https://docs.unidata.ucar.edu/nug/current/getting_and_building_netcdf.html#build_hdf4) in CI.

**Reply by derobins** [reply]
We should do this, but didn't netCDF drop support for the netCDF 2 API? Or can you still build that in?

**Reply by hyoklee** [reply]
I don't know if netCDF dropped support for netCDF 2 API or not.
However, netCDF can read HDF4 files via HDF4's SDS API calls.
I just want to make sure that this proposal doesn't break the current netCDF's capabilities to read HDF4 SDS datasets.
Having this kind of CI will help when this proposal is approved and real work begins.

Overall, I really support your effort to drop the netCDF 2.3.2 API and tools in HDF4.
Historically, HDF4's capabilities of reading old netCDF caused conflict in building netCDF with HDF4 support from source.

**Comment by derobins** [comment]
I feel like people should get the netCDF API from netCDF, not from HDF4.

**Reply by hyoklee** [reply]
Exactly!
